File: src/main/java/no/nav/tjenestepensjon/simulering/config/ExecutorServiceConfig.java
package no.nav.tjenestepensjon.simulering.config;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class ExecutorServiceConfig {

    @Bean
    public ExecutorService taskExecutor(TpLeverandorConfig tpLeverandorConfig) {
        return Executors.newFixedThreadPool(tpLeverandorConfig.tpLeverandorList().size());
    }
}

File: src/main/kotlin/no/nav/tjenestepensjon/simulering/config/ExecutorServiceConfig.kt
package no.nav.tjenestepensjon.simulering.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors

@Configuration
class ExecutorServiceConfig {
    @Bean
    fun taskExecutor(tpLeverandorConfig: TpLeverandorConfig): ExecutorService =
            Executors.newFixedThreadPool(tpLeverandorConfig.tpLeverandorList().size)
}